### DISCUSSION TITLE
Make `int` negative pre-validation

### DISCUSSION THREAD

**Discussion by awgymer** [pydantic/pydantic#11040] [Question]
Is it possible to annotate an attribute with e.g. `NegativeInt` but allow any non-negative ints to try and be converted before that validation?

I thought about using a `before` validator:

```py
@field_validator('mynegativeint', mode='before')
@classmethod
def make_negative(cls, v: Any):
    return -abs(v)
```

This works great if the value passed in is already an `int` (i.e. `4`), but if you try to validate a numeric string `'4'` this will fail because you do not get any underlying type-conversion. 

I'm not sure if this is possible whilst using the `NegativeInt` type-hint (possibly a `wrap` mode validator?) or whether I have to just type-hint as `int` and perform the conversion in an `after` validator?

**Comment by EDohmen** [comment]
How about casting first to `int`?

```python
    @field_validator("mynegativeint", mode="before")
    @classmethod
    def make_negative(cls, mynegativeint: Any, info: dict):
        try:
            mynegativeint = int(mynegativeint)
            return -abs(mynegativeint)
        except ValueError:
            raise ValueError(f"Can't convert {mynegativeint} to negative integer")
```

**Reply by awgymer** [reply]
I did consider this. Is `int(value)` analogous to the way `pydantic` would attempt the type conversion from string internally?
